packages/payload/src/queues/utilities/updateJob.ts

```diff
@@ -0,0 +1,156 @@
+      depth,
+      disableTransaction,
+      req,
+    })
+    if (returning === false) {
+      return null
+    }
+    return result as BaseJob
+  }
+
+  let updateData = sanitizeUpdateData({ data })
+
+  // update updateData to support partial updates
+  // TODO: this can be optimized in the future - partial updates are supported in mongodb. In postgres,
+  // we can support this by manually constructing the sql query
+  const currentJob = await req.payload.db.findOne({
+    collection: jobsCollectionSlug,
+    req: disableTransaction === true ? undefined : req,
+    where: {
+      id: {
+        equals: id,
+      },
+    },
+  })
+
+  if (currentJob) {
+    updateData = {
+      ...currentJob,
+      ...updateData,
+    }
+  }
+
+  const updatedJob = (await req.payload.db.updateOne({
+    id,
+    collection: jobsCollectionSlug,
+    data: updateData,
+    req: disableTransaction === true ? undefined : req,
+    returning: true,
+  })) as BaseJob
+
+  if (returning === false) {
+    return null
+  }
+
+  const res = jobAfterRead({
+    config: req.payload.config,
+    doc: updatedJob,
+  })
+
+  return res
+}
+
+type RunJobsArgs = {
+  data: Partial<BaseJob>
+  depth?: number
+  disableTransaction?: boolean
+  limit?: number
+  req: PayloadRequest
+  returning?: boolean
+  where: Where
+}
+export async function updateJobs({
+  data,
+  depth,
+  disableTransaction,
+  limit,
+  req,
+  returning,
+  where,
+}: RunJobsArgs): Promise<BaseJob[] | null> {
+  if (depth || req.payload.config?.jobs?.runHooks) {
+    const result = await req.payload.update({
+      collection: jobsCollectionSlug,
+      data,
+      depth,
+      disableTransaction,
+      limit,
+      req,
+      where,
+    })
+    if (returning === false || !result) {
+      return null
+    }
+    return result.docs as BaseJob[]
+  }
+
+  let updatedJobs = []
+
+  // TODO: this can be optimized in the future - partial updates are supported in mongodb. In postgres,
+  // we can support this by manually constructing the sql query. We should use req.payload.db.updateMany instead
+  // of req.payload.db.updateOne once this is supported
+  const jobsToUpdate = await req.payload.db.find({
+    collection: jobsCollectionSlug,
+    limit,
+    req: disableTransaction === true ? undefined : req,
+    where,
+  })
+  if (!jobsToUpdate?.docs) {
+    return null
+  }
+
+  for (const job of jobsToUpdate.docs) {
+    const updateData = {
+      ...job,
+      ...data,
+    }
+    const updatedJob = await req.payload.db.updateOne({
+      id: job.id,
+      collection: jobsCollectionSlug,
+      data: sanitizeUpdateData({ data: updateData }),
+      req: disableTransaction === true ? undefined : req,
+      returning,
+    })
+    updatedJobs.push(updatedJob)
+  }
+
+  if (returning === false || !updatedJobs?.length) {
+    return null
+  }
+
+  const res = updatedJobs.map((updatedJob) => {
+    return jobAfterRead({
+      config: req.payload.config,
+      doc: updatedJob,
+    })
+  })
```

Comment:
We are making an assumption that the afterRead is only used for manipulating the job document. What if I was adding a hook in my job to perform some other function like a slack notification or something unrelated?

Reply:
I would treat _all_ custom payload hooks in the payload jobs collection as unsupported, unless `runHooks` is set to `true`.

I wouldn't wanna make an exception for _just_ the afterRead collections hook - that's not consistent/confusing ("Why does my afterRead hook on the collection work, but the afterRead hook on my field and my afterChange hooks don't run??")

Individual tasks have `onSuccess` and `onFail`, and we could add that on the workflow / job-config level as well.

`jobAfterRead` does not run any custom hooks - it just populates the readable task status field. This is also done in the default afterRead hook for when people run `payload.find()` to get a job, which is why I named it `jobAfterRead`.